dateparser/date.py

```diff
@@ -254,6 +252,40 @@ def _is_valid_date_obj(self, date_obj):
         return True
 
 
+class DateData:
+    """
+    Class that represents the parsed data with useful information.
+    It can be accessed like a dict object.
```

Comment:
I assume this is so that existing code won’t break too much.

If we want to go this route, I think we should consider implementing the whole `MutableMapping` interface, otherwise some dict usages will fail.

However, I would also consider the possibility of _not_ providing a dict-like API, or at least only providing it for setting and getting values, which _may_ help avoid situations were we cannot deprecate fields in a clean way. For example, if user code iterates keys, for example, as we add and remove fields in the future warning about deprecation and keeping a backward-compatible behavior can get messy.

Not a strong opinion, though; just a thought.

Reply:
I've been playing around. It would be really easy to support only setting and getting like a `dict` (I can push the code now), but my intention was to leave at least the __repr__() like a `dict` object (this makes it easier to see the content of the object and would require less effort changing the docs). However, adding the __repr__ like a `dict` could easily make people think that it can be used like a dict: iterating, etc.

What do you think? @Gallaecio

Reply:
Hi @Gallaecio, I changed the __repr__() method and removed some methods, could you review it again?
thanks!